File: balloon/zeromq/src/test/python/subscriber.py
__author__ = 'timoveldt'

import zmq
import time


def main():
    context = zmq.Context()

    sub = context.socket(zmq.SUB)
    sub.connect("tcp://localhost:5559")
    topics = 'AC'
    for topic in topics:
        sub.setsockopt(zmq.SUBSCRIBE, topic)
    print "subscribed to: %r" % topics
    while True:
        time.sleep(0.3)
        while True:
            if sub.poll(timeout=1000):
                print "received", sub.recv_multipart()
            else:
                print "Breaking loop, got nothing..."
                break


if __name__ == "__main__":
    main()

File: balloon/zeromq/src/test/cpp/subscriber.cpp
//
// Created by timoveldt on 6/17/15.
//

#include "zmq.hpp"

#include <iostream>
#include <sstream>
#include <unistd.h>

int main(int argc, char *argv[]) {
    zmq::context_t context(1);

    //  First, connect our subscriber socket
    zmq::socket_t subscriber(context, ZMQ_SUB);
    subscriber.setsockopt(ZMQ_SUBSCRIBE, "A", 1);
    subscriber.setsockopt(ZMQ_SUBSCRIBE, "D", 1);
    std::string topic = "CPP_TOPIC";
    subscriber.setsockopt(ZMQ_SUBSCRIBE, topic.data(), topic.size());
    subscriber.connect("tcp://localhost:5559");

    while (1) {
        zmq::message_t message(100);
        subscriber.recv(&message);

        std::string msg_str(static_cast<char*>(message.data()), message.size());


        std::cout << msg_str << std::endl;

        usleep(250 * 100);
    }


    return 0;
}